### PR TITLE
refactor: Updated notification plugins to use a common interface

### PR DESCRIPTION
**Pull request by bunjiboys**
Updated notification plugins to use a new common interface, to allow users to develop their own notification plugins without having to modify any existing code to enable the plugin.

To send notifications you'll need to create instances of `NotificationContact`s since some notification plugins may use the same format for contacts. For example the Slack notification plugin uses email addresses as well to identify users for DM's, so simply matching against a regular expression would potentially spam people.

Example of how emails used to be sent
```python
# Prep variables
subsystem = 'MySubSystem'
sender = 'foo@bar.com'
subject = 'Cloud Inquisitor Email Test'
body_html = '<h1>Hi from Inquisitor</h1>'
body_text = 'Hi from Inquisitor'

# Old method
from cloud_inquisitor.plugins.notifiers.email import send_email
recipients = ['name@domain.tld']
send_email(subsystem, sender, recipients, subject, body_html, body_text)

# New method
from cloud_inquisitor.utils import send_notification, NotificationContact
recipients = [
    NotificationContact(type='email', value='name@domain.tld'),
]
send_notification(
    subsystem='RequiredTags',
    recipients=recipients,
    subject=subject,
    body_html=body_html,
    body_text=body_text
)
```

The `send_notification` method will automatically handle all installed and enabled types of notifiers. Each notifier plugin now must implement two new class attributes:
* `validation` - A regular expression used to validate each recipient before attempting to notify them
* `notifier_type` - A unique identifying string. This is the value provides as `type` for the `NotificationContact`object

With the new function you can send to multiple types of notifiers at once if so desired.

```python
from cloud_inquisitor.utils import send_notification, NotificationContact
recipients = [
    NotificationContact(type='email', value='name@domain.tld'),
    NotificationContact(type='slack', value='#notifications'),
]
send_notification(
    subsystem='RequiredTags',
    recipients=recipients,
    subject=subject,
    body_html=body_html,
    body_text=body_text
)
```

Using the old methods for notification will now trigger a warning in the logs stating the methods have been deprecated. These functions will eventually be removed from the code in a future version

```
[15:12:34] cloud_inquisitor.wrappers send_message has been deprecated, use cloud_inquisitor.utils.send_notifications instead
```

### CODE REVIEW

File: backend/cloud_inquisitor/app.py [hunk 1]
```diff
@@ -211,6 +213,19 @@ def __register_types(self):
...
             logger.warning('Failed loading type information: {}'.format(ex))
+
+    def __register_notifiers(self):
+        """Lists all notifiers to be able to provide metadata for the frontend
+
+        Returns:
+            `list` of `dict`
+        """
+        notifiers = {}
+        for ep in CINQ_PLUGINS['cloud_inquisitor.plugins.notifiers']['plugins']:
```
Comment (rnikoopour) [inline backend/cloud_inquisitor/app.py]: What's ep?

Comment (bunjiboys) [reply backend/cloud_inquisitor/app.py]: Entry Points

Comment (rnikoopour) [reply backend/cloud_inquisitor/app.py]: Could we change the variable name please?  It's not immediately clear in this context like it is when you have `for ep in blah['foo']['entry_points']`.

Comment (bunjiboys) [reply backend/cloud_inquisitor/app.py]: If we want to change that, we can do a refactor for clarity later, but this is an identical pattern we are using in quite a few places, so I'd prefer that to not be changed in one place, but rather be done as a refactor later

Comment (rnikoopour) [reply backend/cloud_inquisitor/app.py]: I hear you, but as I mentioned above, the other cases explicitly call out `entry_points` in their `for in` construct.  This is not clear and can lead to confusion for others.

**Pull request merged**